packages/pwa-kit-extension-sdk/src/types.ts

```diff
@@ -36,3 +38,12 @@ export type BuildCandidatePathsOptions = {
     projectDir: string
     extensionEntries: ApplicationExtensionEntry[]
 }
+
+export interface GetRoutesParams {
+    locals: Record<string, any>
+}
+
+export interface BeforeRouteMatchParams {
+    allRoutes: RouteProps[]
```

Comment:
similar to this [comment](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/2308/files#r2005909799), should allRoutes support both RouteProps and partial routes or `SerializedRoute` if we go with that?